src/nextgen/nextgen_generator.ts

```diff
@@ -11,6 +23,46 @@ export interface Details {
   collection?: number;
 }
 
+export async function listS3Objects(
+  s3: S3Client,
+  path: string
+): Promise<number[]> {
+  const command = new ListObjectsCommand({
+    Bucket: NEXTGEN_BUCKET,
+    Prefix: path
+  });
+  const contents: number[] = [];
+  try {
+    const response = await s3.send(command);
+    response.Contents?.forEach((object) => {
+      if (object.Key) {
+        contents.push(parseInt(object.Key?.replace(path, '')));
+      }
+    });
+  } catch (error) {
+    logger.error(`[S3 LIST ERROR] : [PATH ${path}] : [ERROR ${error}]`);
+  }
+  return contents;
```

Comment:
If an error is caught on line 43, it should return an empty array? Wouldn't we rather want the loop to crash and error propagated to Sentry?

Reply:
contents is empty at this point isn't it?